stdlib/result.gr

```diff
@@ -1,89 +1,201 @@
+ * @returns `Some(okValue)` with okValue representing the value inside of `Ok` if the result is of the `Ok` variant, otherwise returns `None`
+ */
+export let toOption = result => {
+  match (result) {
+    Ok(x) => Some(x),
+    _ => None,
+  }
 }
 
-export let flatMap = (fn, v) => {
-    match (v) {
-        Ok(x)  => fn(x),
-        Err(e) => Err(e)
-    }
+/**
+ * If the Result is `Ok`, applies the given function to the `Ok` value and returns the result. Returns the unmodified `Err` otherwise.
+ *
+ * @param fn: The function to apply
+ * @param result: The result to map
+ * @returns `fn(okValue)` if the result is of the `Ok` variant, okValue represent the value wrapped by the result, otherwise returns the unmodified `Err`
+ */
+export let flatMap = (fn, result) => {
+  match (result) {
+    Ok(x) => fn(x),
+    Err(e) => Err(e),
+  }
 }
 
-export let flatMapErr = (fn, v) => {
-    match (v) {
-        Err(e) => fn(e),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is an `Err`, applies the given function to the `Err` value and returns the result. Returns the unmodified `Ok` value otherwise.
+ *
+ * @param fn: The function to apply
+ * @param result: The result to map
+ * @returns `fn(errValue)` if the result is of the `Err` variant, errValue represent the value wrapped by the result, otherwise returns the unmodified `Ok`
+ */
+export let flatMapErr = (fn, result) => {
+  match (result) {
+    Err(e) => fn(e),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let map = (fn, v) => {
-    match (v) {
-        Ok(x)  => Ok(fn(x)),
-        Err(e) => Err(e)
-    }
+/**
+ * If the Result is `Ok(x)`, returns `Ok(fn(x))`. Returns the unmodified `Err` otherwise.
+ *
+ * @param fn: The function to map
+ * @param result: The result to map on
+ * @returns `Ok(fn(okValue))` if the result is of the `Ok` variant, otherwise returns the unmodified `Err`
+ */
+export let map = (fn, result) => {
+  match (result) {
+    Ok(x) => Ok(fn(x)),
+    Err(e) => Err(e),
+  }
 }
 
-export let mapErr = (fn, v) => {
-    match (v) {
-        Err(e) => Err(fn(e)),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is `Err(x)`, returns `Err(fn(x))`. Returns the unmodified `Ok` otherwise.
+ *
+ * @param fn: The function to map
+ * @param result: The result to map on
+ * @returns `Err(fn(errValue))` if the result is of the `Err` variant, otherwise returns the unmodified `Ok`
+ */
+export let mapErr = (fn, result) => {
+  match (result) {
+    Err(e) => Err(fn(e)),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let mapWithDefault = (fn, def, v) => {
-    match (v) {
-        Ok(a) => fn(a),
-        _     => def
-    }
+/**
+ * If the Result is `Ok(x)`, returns `fn(x)`. Returns the provided default otherwise.
+ *
+ * @param fn: The function to map
+ * @param def: The default value to return
+ * @param result: The result to map on
+ * @returns `Ok(fn(okValue))` if the result is of the `Ok` variant, otherwise returns the value given by the `def` param
```

Comment:
This is wrong. mapWithDefault doesn't need to produce another Result.
```suggestion
 * @returns The value produced by the mapping function if the result is of the `Ok` variant or the default value otherwise 
```